qa/rpc-tests/test_framework/test_framework.py

```diff
@@ -172,12 +173,15 @@ def main(self):
                 # Dump the end of the debug logs, to aid in debugging rare
                 # travis failures.
                 import glob
-                filenames = glob.glob(self.options.tmpdir + "/node*/regtest/debug.log")
+                filenames = [self.options.tmpdir + "/test_framework.log"]
+                filenames += glob.glob(self.options.tmpdir + "/node*/regtest/debug.log")
                 MAX_LINES_TO_PRINT = 1000
                 for f in filenames:
-                    print("From" , f, ":")
-                    from collections import deque
-                    print("".join(deque(open(f), MAX_LINES_TO_PRINT)))
+                    try:
+                        print("From" , f, ":")
+                        print("".join(deque(open(f), MAX_LINES_TO_PRINT)))
+                    except:
```

Comment:
I think this very broad exception handler could hide information we might like to know about. I would remove it, or narrow it, or at least at stick a "traceback.print_exc()" call in here.

If this handler is supposed to handle the case where f does not exist, you could add an "if os.path.exists(f)" or similar check above.